inlong-sort/sort-connectors/jdbc/src/main/java/org/apache/inlong/sort/jdbc/dialect/PostgresDialect.java

```diff
@@ -165,8 +165,9 @@ public List<LogicalTypeRoot> unsupportedTypes() {
     @Override
     public PreparedStatement setQueryPrimaryKeySql(Connection conn,
             String tableIdentifier) throws SQLException {
-        PreparedStatement st = conn.prepareStatement(QUERY_PRIMARY_KEY_SQL);
-        st.setString(1, JdbcMultiBatchingComm.getTableNameFromIdentifier(tableIdentifier));
-        return st;
+        try (PreparedStatement st = conn.prepareStatement(QUERY_PRIMARY_KEY_SQL)) {
+            st.setString(1, JdbcMultiBatchingComm.getTableNameFromIdentifier(tableIdentifier));
+            return st;
+        }
     }
```

Comment:
I don't think here need to use try. AbstractJdbcDialect will close it.

Reply:
restored